File: src/C#/CreationalPatterns/AbstractFactory/EraStyles/Client/Era.cs
using EraStyles.Abstraction;
using System;
using System.Collections.Generic;
using System.Text;

namespace EraStyles.Client
{
    public class Era
    {
        private EraObject _eraObject;

        public Era(EraObjectStylesFactory eraObjectStylesFactory, char era)
        {
            switch (era)
            {
                case 'M':
                    _eraObject = eraObjectStylesFactory.CreateMedievalStyleObject();
                    break;
                case 'R':
                    _eraObject = eraObjectStylesFactory.CreateRenaissanceStyleObject();
                    break;
                case 'V':
                    _eraObject = eraObjectStylesFactory.CreateVictorianEraStyleObject();
                    break;
            }
        }

        public void Info()
        {
            _eraObject.ShowDetails();
        }
    }
}

File: src/JavaScript/CreationalPatterns/AbstractFactory/EraStyles/client/era.js

class Era {
    constructor(stylesFactory, era) {
        switch (era) {
            case 'M':
                this._eraObject = stylesFactory.createMedievalStyleObject();
                break;
            case 'R':
                this._eraObject = stylesFactory.createRenaissanceStyleObject();
                break;
            case 'V':
                this._eraObject = stylesFactory.createVictorianEraStyleObject();
                break;
        }
    }

    info() {
        this._eraObject.showDetails();
    }
}

module.exports = Era;